src/services/codefixes/convertToAsyncFunction.ts

```diff
@@ -189,7 +185,18 @@ namespace ts.codefix {
             }
         });
 
-        return getSynthesizedDeepClone(nodeToRename, /*includeTrivia*/ true, synthNamesMap, lastDotThenMap, checker, originalType);
+        function isCallbackOnTypePromise(child: Node, setOfAllCallbacksToReturn: Map<true>): boolean {
+            let node = child.parent;
+            if (isCallExpression(node) || isIdentifier(node) && !setOfAllCallbacksToReturn.get(getNodeId(node).toString())) {
+                const nodeType = checker.getTypeAtLocation(node);

```

Comment:
Consider extracting a helper like `hasPromiseType(node, checker)` since it seems to be checked in multiple places.